src/RestClient.Net.DependencyInjection/RestClientExtensions.cs

```diff
@@ -1,31 +1,51 @@
 ﻿using Microsoft.Extensions.DependencyInjection;
-using Microsoft.Extensions.Logging;
 using System;
 using System.Net.Http;
 
 namespace RestClient.Net
 {
     public static class RestClientExtensions
     {
-        public static IServiceCollection AddRestClient(this IServiceCollection serviceCollection, Action<CreateClientOptions>? configureClient = null)
+        private static IServiceCollection AddRestClient_Base(this IServiceCollection services)
         {
-            _ = serviceCollection
-            .AddSingleton<CreateHttpClient>((sp) =>
-            {
-                var microsoftHttpClientFactoryWrapper = new MicrosoftHttpClientFactoryWrapper(sp.GetRequiredService<IHttpClientFactory>());
-                return microsoftHttpClientFactoryWrapper.CreateClient;
-            })
-            .AddSingleton<CreateClient>((sp) =>
-            {
-                var clientFactory = new ClientFactory(
-                    sp.GetRequiredService<CreateHttpClient>(),
-                    sp.GetService<ILoggerFactory>());
+            return services
+                .AddSingleton<CreateHttpClient>((sp) =>
+                {
+                    var microsoftHttpClientFactoryWrapper = new MicrosoftHttpClientFactoryWrapper(sp.GetRequiredService<IHttpClientFactory>());
+                    return microsoftHttpClientFactoryWrapper.CreateClient;
+                })
+                .AddSingleton<ClientFactory>()
+                .AddSingleton<CreateClient>((sp) =>
+                {
+                    return sp.GetService<ClientFactory>().CreateClient;
+                });
+        }
+
+        public static IServiceCollection AddRestClient(this IServiceCollection services)
+            => services.AddRestClient("RestClient");
+
+        public static IServiceCollection AddRestClient(this IServiceCollection services, Action<CreateClientOptions> configureClient)
+            => services.AddRestClient("RestClient", configureClient);
+
+        public static IServiceCollection AddRestClient(this IServiceCollection services, Action<CreateClientOptions, IServiceProvider> configureClient)
+            => services.AddRestClient("RestClient", configureClient);
```

Comment:
@DrLeh Ok, yes. I think this is on the right track. Although, I think the name probably should be moved to the CreateClientOptions so we can eliminate a couple of overloads.